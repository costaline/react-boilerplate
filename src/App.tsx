import { useState } from 'react'
import { css } from '@emotion/react'

import '@@/App.css'
import { ReactComponent as ReactLogo } from '@@/assets/react.svg'

function App() {
	const [count, setCount] = useState(0)

	return (
		<div className="App">
			<div>
				<a href="https://vitejs.dev" rel="noreferrer" target="_blank">
					<img alt="Vite logo" className="logo" src="/vite.svg" />
				</a>
				<a href="https://reactjs.org" rel="noreferrer" target="_blank">
					<ReactLogo className="logo react" height="144" width="144" />
				</a>
			</div>
			<h1
				css={css`
					text-decoration: underline;
				`}
			>
				Vite + React
			</h1>
			<div className="card">
				<button onClick={() => setCount((count) => count + 1)}>
					count is {count}
				</button>
				<p>
					Edit <code>src/App.tsx</code> and save to test HMR
				</p>
			</div>
			<p className="read-the-docs">
				Click on the Vite and React logos to learn more
			</p>
		</div>
	)
}

export default App
