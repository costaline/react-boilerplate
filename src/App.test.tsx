import { render, screen } from '@testing-library/react'

import { App } from '@@/App'

describe('example', () => {
	it('renders', () => {
		render(<App />)

		const title = screen.getByTestId('title')

		expect(title).toHaveTextContent(/Vite \+ React/i)
	})

	it('snapshot', () => {
		const view = render(<App />)

		expect(view).toMatchSnapshot()
	})
})
