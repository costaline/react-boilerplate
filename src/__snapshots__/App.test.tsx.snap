// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .emotion-0 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<body>
    <div>
      <div
        class="App"
      >
        <div>
          <a
            href="https://vitejs.dev"
            rel="noreferrer"
            target="_blank"
          >
            <img
              alt="Vite logo"
              class="logo"
              src="/vite.svg"
            />
          </a>
          <a
            href="https://reactjs.org"
            rel="noreferrer"
            target="_blank"
          >
            <div
              class="logo react"
              height="144"
              width="144"
            />
          </a>
        </div>
        <h1
          class="emotion-0"
          data-testid="title"
        >
          Vite + React
        </h1>
        <div
          class="card"
        >
          <button>
            count is 
            0
          </button>
          <p>
            Edit 
            <code>
              src/App.tsx
            </code>
             and save to test HMR
          </p>
        </div>
        <p
          class="read-the-docs"
        >
          Click on the Vite and React logos to learn more
        </p>
      </div>
    </div>
  </body>,
  "container": .emotion-0 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
    <div
      class="App"
    >
      <div>
        <a
          href="https://vitejs.dev"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Vite logo"
            class="logo"
            src="/vite.svg"
          />
        </a>
        <a
          href="https://reactjs.org"
          rel="noreferrer"
          target="_blank"
        >
          <div
            class="logo react"
            height="144"
            width="144"
          />
        </a>
      </div>
      <h1
        class="emotion-0"
        data-testid="title"
      >
        Vite + React
      </h1>
      <div
        class="card"
      >
        <button>
          count is 
          0
        </button>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR
        </p>
      </div>
      <p
        class="read-the-docs"
      >
        Click on the Vite and React logos to learn more
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
